File: android/app/src/main/java/jagtheora/theora/DecoderContext.java
package jagtheora.theora;

import jagtheora.misc.SimplePeer;
import jagtheora.ogg.OggPacket;

public final class DecoderContext extends SimplePeer {

    public DecoderContext(TheoraInfo arg0, SetupInfo arg1) {
        this.init(arg0, arg1);
        if (this.b()) {
            throw new IllegalArgumentException("");
        }
    }

    public double granuleTime(GranulePos arg0) { return 0d; }

    public int decodePacketIn(OggPacket arg0, GranulePos arg1) { return 0; }

    public int getMaxPostProcessingLevel() { return 0; }

    public long granuleFrame(GranulePos arg0) { return 0L; }

    public int setPostProcessingLevel(int arg0) { return 0; }

    public int setGranulePosition(long arg0) { return 0; }

    public int decodeFrame(Frame arg0) { return 0; }

    public void init(TheoraInfo arg0, SetupInfo arg1) {}

    protected void clear() {}
}

File: src/commonMain/kotlin/jagtheora/theora/DecoderContext.kt
package jagtheora.theora

import jagtheora.misc.SimplePeer
import jagtheora.ogg.OggPacket

expect class DecoderContext(arg0: TheoraInfo?, arg1: SetupInfo?) : SimplePeer {

    fun granuleFrame(arg0: GranulePos?): Long

    override fun clear()

    fun setPostProcessingLevel(arg0: Int): Int

    fun decodePacketIn(arg0: OggPacket?, arg1: GranulePos?): Int

    val maxPostProcessingLevel: Int
        get

    fun granuleTime(arg0: GranulePos?): Double

    fun setGranulePosition(arg0: Long): Int

    fun init(arg0: TheoraInfo?, arg1: SetupInfo?)

    fun decodeFrame(arg0: Frame?): Int
}
